pkg/sql/rowexec/bulk_row_writer.go

```diff
@@ -133,6 +133,21 @@ func (sp *bulkRowWriter) wrapDupError(ctx context.Context, orig error) error {
 
 func (sp *bulkRowWriter) ingestLoop(ctx context.Context, kvCh chan row.KVBatch) error {
 	writeTS := sp.spec.Table.CreateAsOfTime
+
+	// Delete existing span before ingestion to prevent key collisions.
+	if sp.spec.DeleteSpan {
```

Comment:
If this operation is distributed, don't we risk stepping on ourselves? Should this be done at a higher level?

Reply:
Yeah, I don't think we should be doing this in the processor; I think we should be doing this in the coordinator before we setup the flow, since a) now every proc is deleting the same span, which will all the levels of the LSM with duplicate range keys pushing all writes into L0, and more importantly b) one proc might already be writing by the time another starts up, gets scheduled and runs its deletes, leading to data corruption.